### ISSUE TITLE
new nvme ssd intel rst problem

### ISSUE THREAD

**Issue by greatkim104** [Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh#5]
### Zenbook Pro Duo Variant

i7/i9 UX581GV (9th-gen with RTX 2060)

### Bios Version

309

### OpenCore Version

[1.0.0](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/releases/tag/0.0.0)

use your EFI

### macOS Version

Ventura (macOS 13)

### Current Behavior

I bought a new nvme ssd (crucial P3 Plus 500GB) to hackintosh UX581GV (i7 model).

after swapping in a new nvme ssd, my laptop gets stuck on 'asus' logo (i cannot access the bios menu)

so I removed the new nvme ssd, after accessing the bios and changing the AHCI to Intel RST premium with acceleration, I can boot from usb with opencore installed, but the nvme ssd is not visible in disk utility. 

i had same problem with a Western Digital SN770 250gb.

I would really appreciate it if you could help me out.

sorry for poor english 

### Expected Behavior

booting from opencore installed usb

installing mac os ventura on crucial p3 plus nvme ssd

### Steps To Reproduce

[BIOS setting with Intel RST Premium with acceleration]
i can boot opencore installed usb.
but, i can't see my nvme ssd on disk utility (only visible opencore installed usb)

[BIOS setting with AHCI]
i can't boot opencore installed usb.
and i also can't access bios
so i have to remove the nvme ssd and change the AHCI in the bios to intel rst premium with acceleration.

**Comment by Qonfused** [comment]
From what it sounds like, you're unable to enter the BIOS with the Crucial SSD and Intel RST disabled? The only action you'd need to take to disable Intel RST is to set the SATA configuration to AHCI in the BIOS -- but it sounds like you've already tried this though.

In Disk Utility, the Intel SSD won't show up when the SATA configuration is set to Intel RST (Optane) instead of SATA -- [this part is expected](https://dortania.github.io/OpenCore-Install-Guide/troubleshooting/extended/opencore-issues.html#can-t-see-macos-partitions).

With the Crucial SSD, I'd see if you can press 'Esc' to enter the boot menu and 'Enter Setup' (BIOS) or boot into OpenCore (likely named 'UEFI OS').

As an aside, if you swap back to Windows on the original SSD, you'll need to re-enable Intel RST, boot into safe mode at least once, and then restart to boot into Windows normally. That's the safest way to ensure that RST is restored correctly.

**Comment by greatkim104** [comment]
> In Disk Utility, the Intel SSD won't show up when the SATA configuration is set to Intel RST (Optane) instead of SATA -- [this part is expected](https://dortania.github.io/OpenCore-Install-Guide/troubleshooting/extended/opencore-issues.html#can-t-see-macos-partitions).
>
Yes, I already read about intel optane disk and intel rst are not working in opencore

>With the Crucial SSD, I'd see if you can press 'Esc' to enter the boot menu and 'Enter Setup' (BIOS) or boot into OpenCore (likely named 'UEFI OS').
>
If i turn on "Intel RST" in bios, and booting with your method(press esc key), the ssd is not recognized by disk utility as shown in the picture

And If i switch back "AHCI" in bios, press esc key in "ASUS" logo screen method is not working in my situation. Stuck on "ASUS" logo


![IMG_5187](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/assets/170222924/cbb533e3-54ab-4d25-80fe-3091e5e0c32f)

![IMG_5188](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/assets/170222924/156a3dd2-5b58-466f-8863-eae43f12e9e0)

Picture "Intel RST", with pressing esc key boot method 

Here is my bios picture
![IMG_5186](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/assets/170222924/3eec3475-b6d6-45a5-ab76-0fbb2fc427da)

**Comment by Qonfused** [comment]
I'd see if you can find and disable an option for VMD in your BIOS (may fall under PCI/IO configuration). My impression is that Intel RST isn't being fully disabled -- I don't see any way it'd halt POSTing otherwise.

There doesn't appear to be any locks for Intel RST in that BIOS version that would prevent updating it.

**Comment by greatkim104** [comment]
Thank you for response

But i can't find any options about VMD(PCI/IO configuration)

My bios only has SATA Mode Selection that i already have switched to
![IMG_5192](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/assets/170222924/aae6102a-42e6-4b68-bce1-e2615bfdd6f2)
![IMG_5191](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/assets/170222924/3716531e-2f18-4db7-97b5-8fd6eafcab2a)
![IMG_5190](https://github.com/Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh/assets/170222924/439ab54d-90e3-4ed0-a5b4-d8fa19c9ed54)

**Comment by Qonfused** [comment]
I see, so then Intel RST should be disabled when in AHCI SATA mode. I'll have to take a closer look at the BIOS to try and figure out why there's a post issue with RST disabled.

I'm curious if it still doesn't post with the Intel SSD installed w/ Intel RST disabled -- For now, I'd try to avoid booting into Windows with that disabled.

**Comment by greatkim104** [comment]
So you are asking is it possible for a crucial ssd(is it "intel ssd" that you are referring to?) to boot with intel RST disabled?

And i also can't boot windows with AHCI mode.(freezing at ASUS logo) 

:(

**Comment by Qonfused** [comment]
I was asking if it still freezes when you boot with the Intel SSD installed and Intel RST disabled. It sounded like you already tested the Crucial SSD with and without Intel RST enabled, so I'm just verifying that there isn't a firmware issue preventing boot without Intel RST enabled.

**Comment by greatkim104** [comment]
I don't have Intel SSD. I have samsung PM981 nvme ssd from asus, i modified the registry in windows, booted into safe mode and switched Intel RST to AHCI in the bios, which worked.  [Like This](https://help.ubuntu.com/rst/)
But Crucial SSD isn't worked by this method

**Comment by Qonfused** [comment]
Ah I see. The newer ZenBook Duo models (like my own) came with Intel SSDs which confused me there. In any case, I suppose that answers my question. If you do happen to have another SATA M.2 drive, it may work as-is.

I matched the part number against the [P3 Plus data sheet](https://www.crucial.com/content/dam/crucial/ssd-products/p3-plus/flyers/b2c/crucial-p3plus-productflyer-consumer.pdf) which corresponds to the Gen4 model, but to my knowledge, there shouldn't be any issues with it working in Gen3 w/ AHCI SATA mode.

It may be worth creating a Windows PE drive and try booting with the Crucial SSD (in RST/Optane mode). Depending on how old the drive is, that'd also provide an opportunity to upgrade the SSD firmware, though I doubt there'd be a relevant update for it unfortunately.

I'm not sure if those Registry tweaks to iaStorV (the Intel RST service) affect the BIOS configuration; since it's a PCH/chipset feature, I'm expecting it to be fairly integrated but some controls there appear (by my guess) to only be for controlling behavior of the Windows Boot Manager, which isn't particularly helpful here.

**Comment by greatkim104** [comment]
Unfortunately, There is no SSD Firmware Update for [Crucial P3 Plus](https://www.crucial.com/support/ssd-support/p3-support).
